pkg/server/leases/controller.go

```diff
@@ -0,0 +1,49 @@
+package leases
+
+import (
+	"context"
+	"time"
+
+	coordinationv1 "k8s.io/api/coordination/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/component-helpers/apimachinery/lease"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+)
+
+type Controller struct {
+	gcController      *GarbageCollectionController
+	acquireController lease.Controller
+}
+
+func NewController(k8sClient kubernetes.Interface, holderIdentity string, leaseDurationSeconds int32, renewInterval time.Duration, gcCheckPeriod time.Duration, leaseName, leaseNamespace string, leaseLabels map[string]string) *Controller {
+	acquireController := lease.NewController(
+		clock.RealClock{},
+		k8sClient,
+		holderIdentity,
+		leaseDurationSeconds,
+		func() {
+			klog.Errorf("repeat heartbeat error in lease acquisition controller for lease %v", leaseName)
```

Comment:
nit: think this already logs on every failure, no need to log here (can just provide nil for the func)

Reply:
Done.